server/src/main/java/org/candlepin/async/JobBuilder.java

```diff
@@ -14,13 +14,287 @@
+        this.arguments = new HashMap<>();
+        this.restrictions = new HashMap<>();
+    }
+
+    /**
+     * Creates a new JobBuilder for the specified asyncronous job task.
+     *
+     * @param jobClass
+     *  The async job class for which to build a job
+     *
+     * @return
+     *  a new JobBuilder instance for the specified job class
+     */
+    public static JobBuilder forTask(Class<? extends AsyncJob> jobClass) {
+        JobBuilder builder = new JobBuilder();
+
+        return builder.setJobClass(jobClass);
+    }
+
+
+    /**
+     * Sets the class representing the task to be executed for this job.
+     *
+     * @param jobClass
+     *  The async job class representing the task to execute for this job
+     *
+     * @return
+     *  this JobBuilder instance
+     */
+    public JobBuilder setJobClass(Class<? extends AsyncJob> jobClass) {
+        if (jobClass == null) {
+            throw new IllegalArgumentException("jobClass is null");
+        }
+
+        this.jobClass = jobClass;
+        return this;
+    }
+
+    /**
+     * Fetches the async job class to execute when running this job.
+     *
+     * @return
+     *  the async job class to execute when running this job, or null if the job class has not yet
+     *  been set
+     */
+    public Class<? extends AsyncJob> getJobClass() {
+        return this.jobClass;
+    }
+
+    /**
+     * Sets the name of this job.
+     *
+     * @param jobName
+     *  The name to set for this job
+     *
+     * @throws IllegalArgumentException
+     *  if the job name is null or empty
+     *
+     * @return
+     *  this JobBuilder instance
+     */
+    public JobBuilder setJobName(String jobName) {
+        if (jobName == null || jobName.isEmpty()) {
+            throw new IllegalArgumentException("jobName is null or empty");
+        }
+
+        this.name = jobName;
+        return this;
+    }
+
+    /**
+     * Fetches the name of this job. If the job name has not yet been set, this method returns
+     * null.
+     *
+     * @return
+     *  The name of this job, or null if the name has not been set
+     */
+    public String getJobName() {
+        return this.name;
+    }
+
+    /**
+     * Sets the group name for this job.
+     *
+     * @param jobGroup
+     *  The name of the group to set for this job
+     *
+     * @throws IllegalArgumentException
+     *  if the group name is null or empty
+     *
+     * @return
+     *  this JobBuilder instance
+     */
+    public JobBuilder setJobGroup(String jobGroup) {
+        if (jobGroup == null || jobGroup.isEmpty()) {
+            throw new IllegalArgumentException("jobGroup is null or empty");
+        }
+
+        this.group = jobGroup;
+        return this;
+    }
+
+    /**
+     * Fetches the name of this job. If the job name has not yet been set, this method returns
+     * null.
+     *
+     * @return
+     *  The name of this job, or null if the name has not been set
+     */
```

Comment:
Javadoc references the name attribute, should be the group.

Reply:
Fixed